src/Windows/Avalonia.Win32/OpenGl/Angle/AngleWin32EglDisplay.cs

```diff
@@ -95,21 +96,41 @@ public static unsafe AngleWin32EglDisplay CreateD3D11Display(Win32AngleEglInterf
                 else
                 {
                     if (factory.EnumAdapters1(0, &pAdapter) != 0)
-                        throw new OpenGlException("No adapters found");
+                        ThrowNoAdaptersFound();
                     chosenAdapter = MicroComRuntime.CreateProxyFor<IDXGIAdapter1>(pAdapter, true);
                 }
             }
 
             IntPtr pD3dDevice;
             using (chosenAdapter)
-                DirectXUnmanagedMethods.D3D11CreateDevice(chosenAdapter?.GetNativeIntPtr() ?? IntPtr.Zero,
-                    D3D_DRIVER_TYPE.D3D_DRIVER_TYPE_UNKNOWN,
+            {
+                // https://learn.microsoft.com/en-us/windows/win32/api/dxgi/ns-dxgi-dxgi_adapter_desc1
+                // https://learn.microsoft.com/en-us/windows/win32/api/dxgi/ne-dxgi-dxgi_adapter_flag
+                var isSoftwareAdapter = (chosenAdapter!.Desc1.Flags & 2) == 1;
+                var driverType = isSoftwareAdapter ?
+                    D3D_DRIVER_TYPE.D3D_DRIVER_TYPE_WARP :
+                    D3D_DRIVER_TYPE.D3D_DRIVER_TYPE_HARDWARE;
+
+                var hr = DirectXUnmanagedMethods.D3D11CreateDevice(chosenAdapter?.GetNativeIntPtr() ?? IntPtr.Zero,
+                    driverType,
                     IntPtr.Zero, 0, featureLevels, (uint)featureLevels.Length,
                     7, out pD3dDevice, out _, null);
 
+                if (pD3dDevice == IntPtr.Zero)
+                {
+                    hr = DirectXUnmanagedMethods.D3D11CreateDevice(chosenAdapter?.GetNativeIntPtr() ?? IntPtr.Zero,
+                        driverType,
+                        IntPtr.Zero, 0, featureLevels, (uint)featureLevels.Length,
+                        7, out pD3dDevice, out _, null);
```

Comment:
What is the difference between two calls?

Reply:
It was supposed to call D3D_DRIVER_TYPE_SOFTWARE.
My bad.

-------

As for how that happened.

I have 2 copies of the project on my dual booting machine, one on Windows, one on (Arch) Linux, and it seems I pushed code to GitHub from the wrong copy, as I was also testing on Wine.

Note: There's a lack of a decent filesystem that works on both. `WinBtrfs` eventually corrupts, and `ntfs3` has bugs when writing files fast and constantly requires I run chkdsk every once and again, as it borks the FS. Maybe exFAT, I still need to find out.

Reply:
I fixed that in any case.

Reply:
Can we fall back to D3D_DRIVER_TYPE_UNKNOWN there which was the old behavior? That way we are guaranteed to not break things that did work previously.